src/components/feed/CreateFeed/ChrisFileSelect.tsx

```diff
@@ -0,0 +1,334 @@
+          name: 'ChRIS Files',
+          path: '/',
+          children: visibleTopLevelChildren,
+        }
+      })
+    }
+  }
+
+  // Computes which files/folders are visible, based on the filter
+  computeVisibleChildren(children: ChrisFile[]): ChrisFile[] {
+    const shownChildren = [];
+    for (const child of children) {
+      // if it's a folder with matching children, show the folder
+      if (child.children) {
+        const folderShownChildren = this.computeVisibleChildren(child.children); // get its shown children
+        if (folderShownChildren.length) { // if it  has shown children
+          const folder = {
+            name: child.name,
+            path: child.path,
+            children: folderShownChildren,
+          }
+          shownChildren.push(folder);
+          continue; // do not re-evaluate folder once it's shown
+        }
+      }
+      if (this.matchesFilter(child.name)) { // is file or folder and matches
+        shownChildren.push(child);
+      }
+    }
+    return shownChildren;
+  }
+
+  resetVisibleTree() {
+    this.setState({ visibleTree: this.state.initialTree })
+  }
+
+  normalizeString(str: string) {
+    return str.toLowerCase().trim();
+  }
+
+  matchesFilter(name: string) {
+    return this.normalizeString(name).includes(this.normalizeString(this.state.filter));
+  }
+
+  // generates file name, with match highlighted, for file explorer
+  generateFileName(node: ChrisFile) {
+    const name = node.name;
+    if (!this.state.filter || !this.matchesFilter(name)) {
+      return name;
+    }
+    const matchIndex = this.normalizeString(name).indexOf(this.normalizeString(this.state.filter));
+    const before = name.substring(0, matchIndex);
+    const match = name.substring(matchIndex, matchIndex + this.state.filter.length);
+    const after = name.substring(matchIndex + this.state.filter.length);
+    return (
+      <React.Fragment>
+        {before}
+        <span className="match-highlight">{match}</span>
+        {after}
+      </React.Fragment>
+    );
+  }
+
+  renderTreeNode = (node: ChrisFile) => {
+    let isSelected;
+    const isFolder = !!node.children;
+    if (isFolder) {
+      // there can never be multiple folders with the same path, and folders don't have ids
+      isSelected = !!this.props.files.find(f => f.path === node.path);
+    } else {
+      // but there can be multiple files with the same path
+      isSelected = !!this.props.files.find(f => f.id === node.id);
+    }
+    const icon = isFolder
+      ? (node.collapsed ? <FolderCloseIcon /> : <FolderOpenIcon></FolderOpenIcon>)
+      : <FileIcon />
+    return (
+      <span className="name">
+        <Checkbox
+          checked={isSelected}
+          id={`check-${node.path}`}
+          aria-label=""
+          onChange={isChecked => this.handleCheckboxChange(isChecked, node)}
+        />
+        {icon}
+        {this.generateFileName(node)}
+      </span>
+    )
+  }
+
+  render() {
+    
```

Comment:
try destructuring to clean the names of vars as follows:
const { files, handleFileRemove } = this.props;
const { initialTreeLoaded, visibleTree } = this.state; 
and then you can just reference by names instead of this.state. and this.props.
Try doing this everwhere in the code that it makes sense.